sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/spring/cloud/autoconfigure/monitor/AzureSpringMonitorAutoConfiguration.java

```diff
@@ -81,42 +59,37 @@ private static boolean applicationInsightsAgentIsAttached() {
-            @Override
-            public void customize(AutoConfigurationCustomizer autoConfigurationCustomizer) {
-
-                if (!isNativeRuntimeExecution() && applicationInsightsAgentIsAttached()) {
-                    LOG.warn("The spring-cloud-azure-starter-monitor Spring starter is disabled because the Application Insights Java agent is enabled."
-                        + " You can remove this message by adding the otel.sdk.disabled=true property.");
-                    autoConfigurationCustomizer.addPropertiesCustomizer(OTEL_DISABLE_CONFIG);
-                    return;
-                }
-
-                if (connectionString == null || connectionString.isEmpty()) {
-                    LOG.warn("Unable to find the Application Insights connection string. The telemetry data won't be sent to Azure.");
-                    // If the user does not provide a connection, we disable the export and leave the instrumentation enabled to spot potential failures from
-                    // the instrumentation, in the customer automatic tests for example.
-                    autoConfigurationCustomizer.addPropertiesCustomizer(NO_EXPORT_CONFIG);
-                    return;
-                }
-
-                if (!connectionString.contains("InstrumentationKey=")) {
-                    throw new WrongConnectionStringException();
-                }
-
-                AutoConfiguredOpenTelemetrySdkBuilder sdkBuilder = (AutoConfiguredOpenTelemetrySdkBuilder) autoConfigurationCustomizer;
-
-                HttpPipeline providedHttpPipeline = httpPipeline.getIfAvailable();
-                if (providedHttpPipeline != null) {
-                    AzureMonitorExporterBuilder azureMonitorExporterBuilder = new AzureMonitorExporterBuilder().connectionString(connectionString).pipeline(providedHttpPipeline);
-                    AzureMonitor.customize(sdkBuilder, azureMonitorExporterBuilder);
-                } else {
-                    AzureMonitor.customize(sdkBuilder, connectionString);
-                }
+    AutoConfigurationCustomizerProvider autoConfigurationCustomizerProvider(@Value("${applicationinsights.connection.string:}") String connectionString, ObjectProvider<AzureMonitorExporterBuilder> azureMonitorExporterBuilder) {
+
+        if (!isNativeRuntimeExecution() && applicationInsightsAgentIsAttached()) {
+            LOG.warn("The spring-cloud-azure-starter-monitor Spring starter is disabled because the Application Insights Java agent is enabled."
+                + " You can remove this message by using the otel.sdk.disabled=true property.");
+            return DISABLE_OTEL_CUSTOMER_PROVIDER;
+        }
+
+        AzureMonitorExporterBuilder providedAzureMonitorExporterBuilder = azureMonitorExporterBuilder.getIfAvailable();
+        if (providedAzureMonitorExporterBuilder != null) {
+            if (System.getenv("APPLICATIONINSIGHTS_CONNECTION_STRING") == null && !connectionString.isEmpty()) {
```

Comment:
will spring pass this env var in as `connectionString` with the binding above?

Reply:
Spring will try to populate `connectionString` with an env var or a  property. If it does find something, an empty string will be used: https://github.com/Azure/azure-sdk-for-java/blob/45e3776da6a1dbc5d566201bca9fc805088fe723/sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/spring/cloud/autoconfigure/monitor/AzureSpringMonitorAutoConfiguration.java#L62

The lines 71 and 72 check that the user has configured `AzureMonitorExporterBuilder` bean with a property to configure the connection string. If it's the case, a log is displayed:

https://github.com/Azure/azure-sdk-for-java/blob/45e3776da6a1dbc5d566201bca9fc805088fe723/sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/spring/cloud/autoconfigure/monitor/AzureSpringMonitorAutoConfiguration.java#L73

https://github.com/Azure/azure-sdk-for-java/blob/45e3776da6a1dbc5d566201bca9fc805088fe723/sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/spring/cloud/autoconfigure/monitor/AzureSpringMonitorAutoConfiguration.java#L74

I have also added a check to know if the user has set `applicationinsights.connection.string` as a JVM property: https://github.com/Azure/azure-sdk-for-java/pull/42006/commits/5f096326a6c075db11962ece01bc4ec36ecef13d The only case that does not work is when the user sets the `applicationinsights.connection.string` property from a Spring `.properties` or `.yml` file (the OTel exporter uses `io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties` that is able to retrieve values from JVM system properties or environment variables).

With the current  `AzureMonitorExporterBuilder`  API there is no way to know if the user has programmatically sets a connection string. So, if the user defines an `AzureMonitorExporterBuilder` bean and sets the `applicationinsights.connection.string` property with a Spring `.properties` or `.yml` file, we display a warning saying that the property from a Spring Boot file is not considered.

Reply:
I have replaced the default connnection string value from empty string to null: https://github.com/Azure/azure-sdk-for-java/pull/42006/commits/d90654be692deb064af6c61a0a5d257a5c7d8109 This may make the code easier to read for some people.